wicket-native-websocket/wicket-native-websocket-core/src/main/java/org/apache/wicket/protocol/ws/api/WebSocketResponse.java

```diff
@@ -87,13 +100,34 @@ public void close()
 			{
 				if (text != null)
 				{
-					connection.sendMessage(text.toString());
+					if (asynchronous)
+					{
+						if (timeout > 0)
+						{
+							connection.sendMessageAsync(text.toString(), timeout);
+						}
+						else
+						{
+							connection.sendMessageAsync(text.toString());
+						}
+					}
+					else
+					{
+						connection.sendMessage(text.toString());
+					}
 					text = null;
 				}
 				else if (binary != null)
 				{
 					byte[] bytes = binary.toByteArray();
-					connection.sendMessage(bytes, 0, bytes.length);
+					if (asynchronous)
+					{
+						connection.sendMessageAsync(bytes, 0, bytes.length);
```

Comment:
Why this method does not use the `timeout` ?

Reply:
Overlooked. Fixing, Thanks!

Reply:
Done